tdt-driver/man/xbidchck.1
.TH XBIDCHCK 1
.SH NAME
xbidchck \- detect and display Tucker Davis XBUS devices
.SH SYNOPSIS
.B xbidchck 
.I ioport
[
.B v
]
.SH DESCRIPTION
.PP
.I xbidchck
shows a diagram of detected XBUS racks, complete with devices.

Each row of text in the diagram displays an XB1 rack number, with
four columns corresponding to the four positions in an XB1 rack.
Non-programmable devices, as well as those racks without programmable
devices, will not be shown in this diagram.

.I xbidchck
does a search of each XB1 rack in your system, looking for programmable
devices, such as PA4 Programmable Attenuators, or SD1 Spike Discriminators.
The starts with the rack with the lowest ID number (see 
.BR xbsysid (1)
), and proceeds through the racks with increasing ID numbers until 
one with no programmable devices is encountered.
.PP
Before you can use 
.I xbidchck,
you must set the communications baud rate with 
.BR xbcomini (1),
and the rack ID's with 
.B xbsysid.
.SH OPTIONS
.I ioport
.RS
This parameter is required. It is the port through which your computer 
communicates with the XBUS, and must be one of "com1", "com2", com3", "com4",
"apa" or "apb".

"com1" through "com4" refer to the physical ports COM1 through COM4.  Use "apa" if you
have an AP2 Array Processor board installed with jumper JP1 set to 
option A (board uses ports 220-238H), and "apb" if JP1 was set to 
option B (board uses ports 240-250H).
.RE
.I v
.RS
Give microcode version information for each device.
.RE

.SH FILES
None.
.SH ENVIRONMENT
None.
.SH DIAGNOSTICS
None.
.SH BUGS
Should be made to detect which port to use via the environment
variable XBCOM, like the other utilities.
.SH AUTHOR
Tucker Davis Technologies, Inc.  <quikki@tdt-quikki.com> 
Adapted to Linux by Chris Malek <cmalek@etho.caltech.edu>
.SH "SEE ALSO"
.BR xbsysid (1),
.BR xbcomini (1),
.BR apld (1),




iserver/Changes
1.2.1:
	tdtproc now checks the environment variable DAQ_AP2_PORT to determine
	which port to use for apinit(); defaults to 'APa'
1.2.2:
	forgot to do the above for XB1init().  Now it does this.
1.2.3:
	made is-matlab.c delete old is_info array before inserting new one
	fixed " characters in set_att.m
1.2.4:
	fixed bug in is-matlab.c:load(): was only copying part of the input
	  waveform to the DA buffer.
	changed is_matlab.c:get_field_int() to check if the field exists
	  before trying to read it, so is_init will not segfault now when
	  you pre-initialize some fields (e.g. is_init.debugflag)
1.2.5:
	glibc > 2.0 does something screwy with stat(), so the call to stat()
	in is-tdt.c:file_is_executable() has been #ifdef'd out
1.2.6:
	in tdtproc.c, removed some extremely uneccesary calls from arm_equipment()
	which were simply chewing up time: program_da1() and check_da_buflen().

	also in tdtproc.c:arm_equipment(), fixed case in ad arming switch() 
1.2.7:
	changed att() in is-matlab: now only takes 1x2 array as its 
	argument, instead of 'left' and 'right' type settings.
1.2.8:
	documented the is_*.m files better, fixed is_record.m
	fixed an error message in is-tdt.c
1.2.9:
	iserver now uses GNU libtool to compile as a shared library on
	both SunOS and Linux.
1.2.10:
	added LED code, removed sync_pulse dependency on ET1
1.2.11:
	Moved matlab install directory to /usr/local/local-matlab

	Release 2:
	Recompiled mex files for Matlab 6
1.2.12:
	Fixed tdtproc/iserver_test.c so that it actually works.
	iserver_test is now part of the distributed package.
	Added code to tdtproc to write DA bufs to files, both before
		and after writing to the AP2 play buffers (DAQ_DA_DUMP and 
		DAQ_AP2_DA_DUMP)


tdt-driver/utils/xbidchck.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <dos.h>
#include <assert.h>

#ifdef MSDOS
#include <conio.h>
#define out(a)		fputs(a, stdout)
#endif

#ifdef NCURSES
#include <curses.h>
#define clrscr()	clear()
#define	gotoxy(x,y)	move((y)-1,(x)-1)
#define kbhit()	        (getch() != ERR)
#define out(a)		addstr(a),refresh()
#endif

#include "xbdrv.h"

int
main(int argc, char *argv[])
{
	int i,j,rn,pn,xbl;

	int 	cc;
	char s[20];
	int vflag,v;


	vflag = 0;
	cc = USE_DOS;

	if (argc > 1) {
		strncpy(s,argv[1], 19);

		if (!strncmp(s,"COM1",4) || !strncmp(s,"com1",4))
			cc = COM1;

		if (!strncmp(s,"COM2",4) || !strncmp(s,"com2",4))
			cc = COM2;

		if (!strncmp(s,"COM3",4) || !strncmp(s,"com3",4))
			cc = COM3;

		if (!strncmp(s,"COM4",4) || !strncmp(s,"com4",4))
			cc = COM4;

		if (!strncmp(s,"APa",3) || !strncmp(s,"apa",3) || !strncmp(s,"APA",3))
			cc = COM_APa;

		if (!strncmp(s,"APb",3) || !strncmp(s,"apb",3) || !strncmp(s,"APB",3))
			cc = COM_APb;


		if ((argc==2) && (!strncmp(argv[1],"V",1) || !strncmp(argv[1],"v",1))) 
			 vflag = 1;
		else 
			if ((argc==3) && (!strncmp(argv[2],"V",1) || !strncmp(argv[2],"v",1)))
				vflag = 1;


		if ((cc==USE_DOS) && (vflag == 0)) {
			printf("\n\n  Illegal argument specified. ");
			printf("\n\n	    Usage:  XBIDCHCK {port_spec} {v}");
			printf("\n	    port_spec = APa, APb, COM1, COM2, COM3 or COM4");
			printf("\n		    v = include microcode version information");

			exit(0);
		}


	}


#ifdef NCURSES

    initscr();
    cbreak();
    noecho();
    /* nodelay(stdscr, TRUE); */

#endif
	clrscr();

	out("XBUS Device Identification Check-Out Program: \n\n");


	XB1init(cc);
	out("    Rack			       Position No.\n");
	out("     No.	 - 1 -		  - 2 -		   - 3 -	    - 4 -\n");
	out("  ----------------------------------------------------------------------------");


	for (i=1; i<=XB_MAX_DEV_TYPES; i++)
	{
		for (j=1; j<=XB_MAX_OF_EACH_TYPE; j++)
		{
			xbl = xbcode[i][j][xbsel];
			if (xbl != 0)
			{
				rn = xbl >> 2;
				pn = xbl & 3;

				gotoxy(5,rn+5);
				sprintf(s,"%d ->",rn);
				out(s);

				if (!vflag)
				{
					gotoxy((pn-1)*17+30,rn+5);
					sprintf(s, "  { %s_(%d) }",xbname[i],j);
					out(s);
				}
				else
				{
					v = XB1version(i,j);
					gotoxy((pn-1)*17+30,rn+5);
					sprintf(s, "{ %s_(%d) v%d }",xbname[i],j,v);
					out(s);
				}
			}
		}
	}


  gotoxy(15,20);
  out("Press any key to continue...\n");
  while (!kbhit());


  endwin();
}
